server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -117,6 +117,7 @@ public int get() {
     private final SeqNoFieldMapper.SequenceIDFields seqID;
     private final Set<String> fieldsAppliedFromTemplates;
     private final Set<String> copyToFields;
+    private boolean sourceStored = false;
```

Comment:
For my understanding - this means "source is already stored at a higher level of object hierarchy"?

Reply:
Right, _ignored_source should no longer be used for this parse subcontext.

Reply:
Maybe rename this to `clonedSource`? Since this is set to true when invoking `cloneDocumentParserContext(...)` method?

Reply:
We do store the source in `_ignored_source` though, no? The point here is to relay that we shouldn't be writing to `_ignored_source` anything for the current parsing context, since it's already covered.

Reply:
Maybe we can see that as a side effect of the cloning of a `DocumentParserContext`?